src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,137 @@
+import { useCallback, useEffect, useState } from 'react';
+import { Status } from '../../types/Status';
+import { Todo } from '../../types/Todo';
+import { TodoList } from '../TodoList';
+import { TodosFilter } from '../TodosFilter';
+
+type Props = {
+  todos: Todo[],
+  setTodos: (todos: Todo[]) => void,
+};
+
+export const TodoApp: React.FC<Props> = ({ todos, setTodos }) => {
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [activeTodos, setActiveTodos] = useState(
+    todos.reduce((acc, curr) => acc + (curr.completed ? 0 : 1), 0),
+  );
+  const [filterStatus, setFilterStatus] = useState(Status.All);
+  const [filteredTodos, setFilteredTodos] = useState([...todos]);
```

Comment:
Don't save filtered todos in state, save only filter value.
Remember about single source of truth rule. Todos here should come only from props.